nemo/collections/nlp/models/language_modeling/megatron_t5_speechlm_pretrain_model.py

```diff
@@ -105,6 +106,21 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         speech_offset = cfg.data.get('speech_offset', 30000)
         speech_head_type = cfg.get('speech_head_type', 'token_level')  # token_level, linear
 
+        self.speech_offset = speech_offset
```

Comment:
why is there two different T5 models? One for pretrain and one for SFT I am assuming. Why does the model need to be defined separately?

Reply:
Yes, as discussed in Slack, these two files can potentially be merged in the future once the two models are stable.

Reply:
@paarthneekhara You should consider inheriting as many common functions as you can, and only overwrite those that you are changing

Reply:
Yes, I think right now these two files became substantially different. The pretraining model is more like megatron_gpt_model.py and SFT model is more like megatron_gpt_prompt_learning.py . But yes, once we finalize the two models we can discuss how to merge them into one.